### ISSUE TITLE
videochat2的demo结果是乱码

### ISSUE THREAD

**Issue by snowyrain** [OpenGVLab/Ask-Anything#113]
按照demo.ipynb跑出来的结果是乱码，josn的预训练模型都是按照提示的方式导入，没有出现报错，请问这是什么原因呀。
demo跑出的结果如下：
###Human: <Video><VideoHere></Video>
###Human: Describe the video in details.
###Assistant:
Act十pub Bohexpected</s>

**Comment by timeparticle** [comment]
我这别也是这样哎

**Comment by Lastsource** [comment]
请问你解决了吗？我处理llama权重过后还是乱码

**Comment by Andy1621** [comment]
大概率是Vicuna-v0权重的问题，可以试试我们新的Mistral版本的模型，权重比较好处理

**Comment by J1nXx136** [comment]
> 请问你解决了吗？我处理llama权重过后还是乱码

请问解决了吗，我目前也是这样

**Comment by yinanhe** [comment]
@J1nXx136 是用llama1的权重转的吗

**Comment by J1nXx136** [comment]
> @J1nXx136 是用llama1的权重转的吗

llama2

**Comment by yinanhe** [comment]
> > @J1nXx136 是用llama1的权重转的吗
> 
> llama2

@J1nXx136  需要使用llama1的权重配合vicuna的delta权重转哦~ 另外，我们推荐体验性能更加好的VideoChat2-HD~
https://github.com/OpenGVLab/Ask-Anything/blob/main/video_chat2/demo/demo_mistral_hd.ipynb

**Comment by bozagina** [comment]
> > > @J1nXx136 是用llama1的权重转的吗
> > 
> > 
> > llama2
> 
> @J1nXx136 需要使用llama1的权重配合vicuna的delta权重转哦~ 另外，我们推荐体验性能更加好的VideoChat2-HD~ https://github.com/OpenGVLab/Ask-Anything/blob/main/video_chat2/demo/demo_mistral_hd.ipynb

您好，现在使用llama-7b，使用vicuna-7b-delta-v0转换，会出现The size of tensor a (32000) must match the size of tensor b (32001) at non-singleton dimension 0，使用v1.1的可以吗

**Comment by J1nXx136** [comment]
> > > > @J1nXx136 是用llama1的权重转的吗
> > > 
> > > 
> > > llama2
> > 
> > 
> > @J1nXx136 需要使用llama1的权重配合vicuna的delta权重转哦~ 另外，我们推荐体验性能更加好的VideoChat2-HD~ https://github.com/OpenGVLab/Ask-Anything/blob/main/video_chat2/demo/demo_mistral_hd.ipynb
> 
> 您好，现在使用llama-7b，使用vicuna-7b-delta-v0转换，会出现The size of tensor a (32000) must match the size of tensor b (32001) at non-singleton dimension 0，使用v1.1的可以吗

我的解决方法是apply_delta.py 行号137处做以下修改：
print("Applying the delta")
    for name, param in tqdm(base.state_dict().items(), desc="Applying delta"):
        assert name in delta.state_dict()
        # param.data += delta.state_dict()[name]
        param.data = torch.cat((param.data, delta.state_dict()[name]), dim=0)

**Comment by bozagina** [comment]
> > > > > @J1nXx136 是用llama1的权重转的吗
> > > > 
> > > > 
> > > > llama2
> > > 
> > > 
> > > @J1nXx136 需要使用llama1的权重配合vicuna的delta权重转哦~ 另外，我们推荐体验性能更加好的VideoChat2-HD~ https://github.com/OpenGVLab/Ask-Anything/blob/main/video_chat2/demo/demo_mistral_hd.ipynb
> > 
> > 
> > 您好，现在使用llama-7b，使用vicuna-7b-delta-v0转换，会出现The size of tensor a (32000) must match the size of tensor b (32001) at non-singleton dimension 0，使用v1.1的可以吗
> 
> 我的解决方法是apply_delta.py 行号137处做以下修改： print("Applying the delta") for name, param in tqdm(base.state_dict().items(), desc="Applying delta"): assert name in delta.state_dict() # param.data += delta.state_dict()[name] param.data = torch.cat((param.data, delta.state_dict()[name]), dim=0)

有用，非常感谢！

**Comment by bozagina** [comment]
> > > > > @J1nXx136 是用llama1的权重转的吗
> > > > 
> > > > 
> > > > llama2
> > > 
> > > 
> > > @J1nXx136 需要使用llama1的权重配合vicuna的delta权重转哦~ 另外，我们推荐体验性能更加好的VideoChat2-HD~ https://github.com/OpenGVLab/Ask-Anything/blob/main/video_chat2/demo/demo_mistral_hd.ipynb
> > 
> > 
> > 您好，现在使用llama-7b，使用vicuna-7b-delta-v0转换，会出现The size of tensor a (32000) must match the size of tensor b (32001) at non-singleton dimension 0，使用v1.1的可以吗
> 
> 我的解决方法是apply_delta.py 行号137处做以下修改： print("Applying the delta") for name, param in tqdm(base.state_dict().items(), desc="Applying delta"): assert name in delta.state_dict() # param.data += delta.state_dict()[name] param.data = torch.cat((param.data, delta.state_dict()[name]), dim=0)

请问videochat2也是用这个vicuna-7b-v0吗，生成出来的内容效果极差...

**Comment by yinanhe** [comment]
> > > > > > @J1nXx136 是用llama1的权重转的吗
> > > > > 
> > > > > 
> > > > > llama2
> > > > 
> > > > 
> > > > @J1nXx136 需要使用llama1的权重配合vicuna的delta权重转哦~ 另外，我们推荐体验性能更加好的VideoChat2-HD~ https://github.com/OpenGVLab/Ask-Anything/blob/main/video_chat2/demo/demo_mistral_hd.ipynb
> > > 
> > > 
> > > 您好，现在使用llama-7b，使用vicuna-7b-delta-v0转换，会出现The size of tensor a (32000) must match the size of tensor b (32001) at non-singleton dimension 0，使用v1.1的可以吗
> > 
> > 
> > 我的解决方法是apply_delta.py 行号137处做以下修改： print("Applying the delta") for name, param in tqdm(base.state_dict().items(), desc="Applying delta"): assert name in delta.state_dict() # param.data += delta.state_dict()[name] param.data = torch.cat((param.data, delta.state_dict()[name]), dim=0)
> 
> 请问videochat2也是用这个vicuna-7b-v0吗，生成出来的内容效果极差...

这个取决于您下载的videochat2的权重是哪个哈~ vc2 含有Phi3/vicuna/mistral的不同版本

**Comment by yinanhe** [comment]
> > > > @J1nXx136 是用llama1的权重转的吗
> > > 
> > > 
> > > llama2
> > 
> > 
> > @J1nXx136 需要使用llama1的权重配合vicuna的delta权重转哦~ 另外，我们推荐体验性能更加好的VideoChat2-HD~ https://github.com/OpenGVLab/Ask-Anything/blob/main/video_chat2/demo/demo_mistral_hd.ipynb
> 
> 您好，现在使用llama-7b，使用vicuna-7b-delta-v0转换，会出现The size of tensor a (32000) must match the size of tensor b (32001) at non-singleton dimension 0，使用v1.1的可以吗

使用v1.1是不可以的，当时的模型只和v0做了对齐
